packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -961,476 +923,437 @@ describe('Dropdown', () => {
+        multiple: true,
+      })
 
-      triggerButton.simulate('click')
-      const itemsList = getItemsListWrapper(wrapper)
-      itemsList
-        .simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
-        .simulate('keydown', { keyCode: keyboardKey.Tab, key: 'Tab', preventDefault })
+      keyDownOnItemsList('Tab', { preventDefault })
 
       expect(preventDefault).toBeCalled()
     })
 
     it('keeps focus on input when not passed', () => {
-      const wrapper = mountWithProvider(<Dropdown multiple search items={items} />)
-      const toggleIndicator = getToggleIndicatorWrapper(wrapper)
+      const { keyDownOnSearchInput } = renderDropdown({
+        defaultOpen: true,
+        defaultHighlightedIndex: 0,
+        multiple: true,
+        search: true,
+      })
 
-      toggleIndicator.simulate('click')
-      const searchInput = getSearchInputWrapper(wrapper)
-      searchInput
-        .simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
-        .simulate('keydown', { keyCode: keyboardKey.Tab, key: 'Tab', preventDefault })
+      keyDownOnSearchInput('Tab', { preventDefault })
 
       expect(preventDefault).toBeCalled()
     })
 
-    it('allows focus to move to next item when passed', () => {
-      const wrapper = mountWithProvider(<Dropdown multiple items={items} moveFocusOnTab />)
-      const triggerButton = getTriggerButtonWrapper(wrapper)
+    it('allows focus to move to next item from search input when passed', () => {
+      const { keyDownOnSearchInput } = renderDropdown({
+        defaultOpen: true,
+        defaultHighlightedIndex: 0,
+        multiple: true,
+        search: true,
+        moveFocusOnTab: true,
+      })
+
+      keyDownOnSearchInput('Tab', { preventDefault })
+
+      expect(preventDefault).not.toBeCalled()
+    })
+
+    it('allows focus to move to next item from items list when passed', () => {
+      const { keyDownOnItemsList } = renderDropdown({
+        defaultOpen: true,
+        defaultHighlightedIndex: 0,
+        multiple: true,
+        moveFocusOnTab: true,
+      })
 
-      triggerButton.simulate('click')
-      const itemsList = getItemsListWrapper(wrapper)
-      itemsList
-        .simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
-        .simulate('keydown', { keyCode: keyboardKey.Tab, key: 'Tab', preventDefault })
+      keyDownOnItemsList('Tab', { preventDefault })
 
       expect(preventDefault).not.toBeCalled()
     })
   })
 
   describe('multiple', () => {
     it('can be switched to "multiple"', () => {
-      const wrapper = mountWithProvider(<Dropdown items={items} value={items[0]} />)
+      const { wrapper, getSelectedItems } = renderDropdown({ value: items[0] })
 
-      expect(
-        findIntrinsicElement(wrapper, `.${Dropdown.slotClassNames.selectedItem}`),
-      ).toHaveLength(0)
+      expect(getSelectedItems()).toHaveLength(0)
 
       wrapper.setProps({ multiple: true } as any)
```

Comment:
```suggestion
      wrapper.setProps({ multiple: true })
```

nit: Do we still need `as any` there?

Reply:
Nope, thanks! But if we revert the changes in `withProvider` then we need them again.